### ISSUE TITLE
Changing i18n provider multiple times is not reflected in the front end

### ISSUE THREAD

**Issue by vaadin-miki** [vaadin/vaadin-date-picker#743]
There seems to be some caching involved when the calendar is opened. For example, weekday names and first day of week are never refreshed and are displayed with the data that was passed to the component before the calendar was first shown.

Similarly, after changing the i18n provider and opening the calendar, the month names change only after scrolling about three, four months into the future or past.

Here is when the i18n provider was set to English, the calendar was opened, then the change to Chinese happened and I scrolled a few months:

![obraz](https://user-images.githubusercontent.com/1917965/79757306-96c86c00-8324-11ea-93b0-068017bc69c9.png)

And here is when the i18n provider was set to Chinese and then the calendar was opened:
![obraz](https://user-images.githubusercontent.com/1917965/79758462-1d317d80-8326-11ea-825f-21a29aae55c1.png)

**Comment by web-padawan** [comment]
Closing in favor of #731 which has some code examples and because both issues are now in the same repository.
